### ISSUE TITLE
No longer persist: postISRCCD, deepCoadd_directWarp, deepCoadd_psfMatchedWarp

### ISSUE THREAD

**Issue by wmwv** [LSSTDESC/ImageProcessingPipelines#92]
I propose that we no longer persist the following intermediate products:

datasetType | stage | Location on disk within rerun
-------------|-------|----------
postISRCCD | calexp | `postISRCCD/<visit>-<filter>/<raftName>/postISRCCD-*`
deepCoadd_directWarp | coadd | `deepCoadd/<filter>/<tract>/<patch>/warp-*` files
deepCoadd_psfMatchedWarp | coadd | `deepCoadd/<filter>/<tract>/<patch>/psfMatchedWarp-*`

If we have cause to do specific investigation runs that do persist these to do various test we can do so, but our default should be not to persist.

**Comment by heather999** [comment]
@wmwv are any of these products needed after calexps such as during coadds or multiband, etc?  Or are these really intermediate during the production of the calexps?
Is it clear how to turn these off or is it just a matter of cleaning up after running SingleFrameDriver?

**Comment by wmwv** [comment]
`postISRCCD` is an intermediate product of `processCcd.py` (i.e., the 'calexp' stage)

`deepCoadd_directWarp` and `deepCoadd_psfMatchedWarp` are intermediate products of the 'coadd' stage.

**Comment by heather999** [comment]
Apparently `postISRCCD` is no longer written by default for hsc, but we need to prod DM for similar changes for `obs_lsst`.  I'll try to follow this and see where it goes: https://lsstc.slack.com/archives/C2JPMCF5X/p1555439400008300

**Comment by wmwv** [comment]
I don't think this sort of configuration should necessarily be in `obs_lsst`.
DESC should plan to explicit configure this in our pipeline workflows.

Whether or not to persist particular products is completely an issues of how you want to do processing.  It's not inherent to a camera or even to a particular survey.

**Comment by heather999** [comment]
Sure, but we may use DM configs to achieve our desired set of persisted data products.  In the case of `postISRCCD`, there is a new JIRA: https://jira.lsstcorp.org/browse/DM-19677, and indeed this would not be something we explicitly add to `obs_lsst` configs.

**Comment by wmwv** [comment]
Yes.  Philosophical issues aside, I do agree that reasonable defaults within the DM Science Pipelines pipe_tasks for `processCcd.py` and the coaddition step should be to not persist these intermediate products.

**Comment by johannct** [comment]
Where do we stand about this. If I read DM-10677 correctly, the switch off is this line `Set IsrTask.ConfigClass.doWrite=False when run as part of ProcessCcdTask;` We run singleFrameDriver and in obs_lsst we already have a config that loads the processCcd config. So should we add it to the obs_lsst branch of the pipeline? @wmwv @heather999 ? I guess the point of not putting it in IPP is just that it is going to be the obs_base/obs_lsst default, but in a too recent release to be used now for DC2 run2.1i

**Comment by heather999** [comment]
Ah this is the JIRA: https://jira.lsstcorp.org/browse/DM-19677 which was included in `w_2019_19`. I will check on the default setting of `IsrTask.ConfigClass.doWrite` and make sure the `dc2/run2.1` branch of `obs_lsst` sets it correctly. This will only take care of `postISRCCD`.  We also want to remove/eliminate `deepCoadd_directWarp` and `deepCoadd_psfMatchedWarp`. We can do so after the fact by explicitly removing those directories.  I can inquire if there is a way to turn off writing these intermediate products too.

**Comment by heather999** [comment]
After pinging the DM folks, it sounds like we cannot turn off the `deepCoadd` intermediate products, so instead we need to remove them afterwards:  https://lsstc.slack.com/archives/C2JPMCF5X/p1559059169022000
I have a script that might help out with that.  It would be best to remove this before transfer to NERSC, but if that's not possible, I can certainly deal with it on this end. @johannct @airnandez who at IN2P3 would be responsible for "cleaning" the data before transfer?

**Comment by airnandez** [comment]
I suggest we decouple the fact of removing the irrelevant data products at IN2P3 from the transfer process.

I propose that we identify and document the files and/or directories produced by the execution of the image processing pipelines that we do not need to be persisted nor transferred to NERSC. That list is presumably composed of the files listed at the [top of this issue](https://github.com/LSSTDESC/ImageProcessingPipelines/issues/92#issue-441724770).

We will then not include those files/directories in the transfer campaigns, independently whether they are present on disk at IN2P3 or not. That allows us to decouple the removal of those files from the transfer of the relevant files to NERSC.

In the [Data Transfer section of the DC2 preparation document](https://github.com/LSSTDESC/ImageProcessingPipelines/wiki/DESC-DC2-Run2.1i-Preparation-Notes-—-DRAFT#step-6-transfer)  the list of files and directories to include in the transfer campaigns IN2P3 → NERSC are already documented. I'm assuming this list is already stabilised. I propose we document there as well the files and directories **to exclude** from the list files to transfer.

As for the question of what tools to use and when to remove those files, I would suggest those files be removed by the image processing pipeline itself, perhaps as a *post-processing* task, similar to the *setup* tasks that are performed before starting each processing stage. I think it would be safer if humans were not involved in this removal process.

**Comment by heather999** [comment]
Thanks, Fabio - that sounds perfect.  I've added the files to exclude to the document.
I will also note that as of `w_2019_19`, the `calexp/postISRCCD` is no longer generated by default. For an added bit of insurance, I can add `config.isr.doWrite=False` to the `obs_lsst/config/imsim/processCcd.py` config file.

**Comment by heather999** [comment]
As discussed at the DM-DC2 meeting today,  `calexp/postISRCCD` is already handled by the configuration and is not longer being stored.  We  also reconfirmed that `deepCoadd_directWarp` and `deepCoadd_psfMatchedWarp` can be removed from the coadd output.  It would be best to handle this as part of the pipeilne or as a post-process before data transfer to NERSC.  When I've done this at NERSC on Run1.2i data, I do: `find coadd-v1/deepCoadd -maxdepth 4 -name "[0-9],[0-9]" -exec rm -rf {} +`

**Comment by johannct** [comment]
I believe that somewhere else, `calexp_camera` was listed as not useful anymore, once skyCorrection is completed. Right?

**Comment by heather999** [comment]
Right, we also confirmed that yesterday at the DM-DC2 meeting.  `calexp_camera` can also be removed from the `calexp` outputs.  I went ahead and did that in the `calexp-v1` copy we have now at NERSC.
